src/librustc_trans/context.rs

```diff
@@ -322,6 +322,7 @@ unsafe fn create_context_and_module(sess: &Session, mod_name: &str) -> (ContextR
     let llvm_target = sess.target.target.llvm_target.as_bytes();
     let llvm_target = CString::new(llvm_target).unwrap();
     llvm::LLVMRustSetNormalizedTarget(llmod, llvm_target.as_ptr());
+    llvm::LLVMRustSetModulePIELevel(llmod);
```

Comment:
Adding the exact same code to calculate `any_library` and `reloc_model` and then gating that call seems to work now, even with `PIELevel::Level::Large` (which, to my understanding should be equivalent to the old option).